File: test/server_lib/translator/node_test.ml
open Sxfiler_core
module D = Sxfiler_domain
module Tr = Sxfiler_server_translator
module T = Sxfiler_rpc.Types

let data =
  { D.Node.id = "id"
  ; full_path = Path.of_string "/bar"
  ; stat =
      D.File_stat.make ~mode:(Int32.of_int 0o775) ~uid:1000 ~gid:1000 ~atime:(Int64.of_int 100)
        ~mtime:(Int64.of_int 1000) ~ctime:(Int64.of_int 10000) ~size:Int64.max_int
        ~is_directory:true ~is_file:false ~is_symlink:true
  ; link_path = Some "/foo" }

let test_set =
  [ ( "can translate to/from domain"
    , `Quick
    , fun () ->
      let expected =
        { T.Node.id = "id"
        ; name = "bar"
        ; stat = Tr.File_stat.of_domain data.stat
        ; parent_directory = "/"
        ; link_path = Some "/foo" }
      in
      Alcotest.(check @@ of_pp Fmt.nop) "domain" expected (Tr.Node.of_domain data) )
  ; ( "can translate to/from yojson"
    , `Quick
    , fun () ->
      let data = Tr.Node.of_domain data in
      Alcotest.(check @@ result (of_pp Fmt.nop) (of_pp Fmt.nop))
        "yojson" (Ok data)
        (Tr.Node.of_yojson @@ Tr.Node.to_yojson data) ) ]

File: src/ts/domains/node.test.ts
import { createNode } from "./node";
import { createFileStat } from "./file-stat";
import { emptyMode } from "./mode";

const stat = createFileStat({
  mode: emptyMode(),
  uid: 1000,
  gid: 1000,
  atime: "10",
  ctime: "11",
  mtime: "12",
  size: "100",
  isDirectory: false,
  isFile: true,
  isSymlink: false,
});

describe("Node value object", () => {
  it("can create the node", () => {
    const node = createNode({
      id: "id",
      name: "file",
      stat,
      parentDirectory: "parent",
      linkPath: "foo",
      marked: false,
    });

    expect(node.id).toEqual("id");
    expect(node.name).toEqual("file");
    expect(node.stat).toEqual(stat);
    expect(node.parentDirectory).toEqual("parent");
    expect(node.linkPath).toEqual("foo");
    expect(node.marked).toBeFalsy();
  });
});
